src/amo/components/SearchForm.js

```diff
@@ -61,8 +77,12 @@ export class SearchFormBase extends React.Component {
   }
 }
 
-export function mapStateToProps({ api }) {
-  return { api };
+export function mapStateToProps(state) {
+  return {
+    addonType: validAddonTypes.includes(state.viewContext.context) ?
```

Comment:
But the `addonType` in this property was only introduced in your patch. Can you give me an example in this component where `addonType` is used for something other than the view context? In every place I see it, it's used as the view context so why not simplify the component (and the tests) by calling it what it is:

```js
return {
  viewContext: state.viewContext.context,
}
```